src/services/TwoFactorService.ts

```diff
@@ -0,0 +1,116 @@
+import { authenticator } from 'otplib';
+import * as QRCode from 'qrcode';
+import { User } from '../entities/User';
+import { initORM } from '../db';
+
+class TwoFactorService {
+  private readonly appName = 'Ghost Drive';
+
+  /**
+   * Generate a new 2FA secret for a user
+   */
+  generateSecret(): string {
+    return authenticator.generateSecret();
+  }
+
+  /**
+   * Generate QR code data URL for 2FA setup
+   */
+  async generateQRCode(username: string, secret: string): Promise<string> {
+    const otpauth = authenticator.keyuri(username, this.appName, secret);
+    return await QRCode.toDataURL(otpauth);
+  }
+
+  /**
+   * Verify a 2FA token
+   */
+  verifyToken(token: string, secret: string): boolean {
```

Comment:
Also for that secret 2fa auth is saved in user table database why the heck user need to send to this API?